src/contracting/storage/orm.py

```diff
@@ -138,4 +148,127 @@ def __getitem__(self, item):
         return super().__getitem__(item)
 
     def clear(self, *args):
-        raise Exception('Cannot write with a ForeignHash.')
+        raise Exception("Cannot write with a ForeignHash.")
+
+
+class LogEvent:
+    """
+    TODO
+    - Break validation into smaller functions
+    - Add checks for use of illegal types and argument names (See Hash checks.)
+    """
+
+    def __init__(self, contract, name, event, params, driver: Driver = driver):
+
+        self._driver = driver
+        self._params = params
+        self._contract = contract
+        self._event = event
+
+        assert isinstance(params, dict), "Args must be a dictionary."
+        assert len(params) > 0, "Args must have at least one argument."
+        # Check for indexed arguments with a maximum of three
+        indexed_args_count = sum(1 for arg in params.values() if arg.get("idx", False))
+        assert (
+            indexed_args_count <= 3
+        ), "Args must have at most three indexed arguments."
+        for param in params.values():
+            if not isinstance(param["type"], tuple):
+                param["type"] = (param["type"],)
+
+            assert all(
+                issubclass(t, (str, int, float, bool))
+                for t in param["type"]
+            ), "Each type in args must be str, int, float, or bool."
+
+
+    def write_event(self, event_data):
+        print(f"event_data: {event_data}")
+        assert len(event_data) == len(
+            self._params
+        ), "Event Data must have the same number of arguments as specified in the event."
+
+        # Check for unexpected arguments
+        for arg in event_data:
+            assert (
+                arg in self._params
+            ), f"Unexpected argument {arg} in the data dictionary."
+
+        # Check for missing and type-mismatched arguments
+        for arg in self._params:
+            assert (
+                arg in event_data
+            ), f"Argument {arg} is missing from the data dictionary."
+
+            # Check the type of the argument
+            assert isinstance(event_data[arg], self._params[arg]["type"]), (
+                f"Argument {arg} is the wrong type! "
+                f"Expected {self._params[arg]['type']}, got {type(event_data[arg])}."
+            )
+
+            # Check the size of the argument
+            value_size = len(str(event_data[arg]).encode("utf-8"))
+            assert (
+                value_size <= 1024
+            ), f"Argument {arg} is too large ({value_size} bytes). Max is 1024 bytes."
```

Comment:
not sure about this. we can probably just not have any limit. user is paying stamps after all.